CI is broken on main

### Background

Currently, CI fails with

```
Run cargo install cargo-nextest
    Updating crates.io index
error: cannot install package `cargo-nextest 0.9.108`, it requires rustc 1.87 or newer, while the currently active rustc version is 1.86.0
`cargo-nextest 0.9.106` supports rustc 1.86
Error: Process completed with exit code 101.
```

see example run: https://github.com/near/threshold-signatures/actions/runs/18804837798/job/53657396860?pr=169

Somewhere we're not pinning versions.

### User Story

As a developer I want CI to be able to execute our tests correctly.

### Acceptance Criteria

CI passes again for any bug-free PRs.

### Resources & Additional Notes

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,7 +41,9 @@ jobs:
           cargo fmt -- --check
 
       - name: Install cargo-nextest
-        run: cargo install cargo-nextest --version 0.9.106
+        uses: taiki-e/install-action@v2
+        with:
+          tool: nextest@0.9.106
 
       - name: Run cargo-nextest
         run: cargo nextest run --locked --all-features --cargo-profile test-release
@@ -63,7 +65,9 @@ jobs:
       - name: Install cargo-llvm-cov
         uses: taiki-e/install-action@cargo-llvm-cov
       - name: Install cargo-nextest
-        run: cargo install cargo-nextest --version 0.9.106
+        uses: taiki-e/install-action@v2
+        with:
+          tool: nextest@0.9.106
       - name: Generate code coverage
         run: cargo llvm-cov nextest --cargo-profile test-release --all-features --workspace --codecov --output-path codecov.json
       - name: Upload coverage to Codecov

